src/services/ipfs-service.js

```diff
@@ -31,22 +37,46 @@ class IpfsService {
 
       this.ipfs.files.add([file], (error, response) => {
         if (error) {
-          console.log("Can't connect to the IPFS Gateway.")
+          console.error("Can't connect to IPFS.")
           console.error(error)
-          reject('Can\'t connect to the IPFS Gateway. Failure to submit listing to Ipfs')
-          return
+          reject('Can\'t connect to IPFS. Failure to submit listing to IPFS')
         }
         const file = response[0]
         const ipfsListing = file.hash
 
         if (ipfsListing) {
           resolve(ipfsListing)
         } else {
-          reject('Failure to submit listing to Ipfs')
+          reject('Failure to submit listing to IPFS')
         }
       })
     });
   }
+
+  getListing(ipfsHashStr) {
+    return new Promise((resolve, reject) => {
+
+      this.ipfs.files.cat(ipfsHashStr, function (err, stream) {
+        if (err) {
+          console.error(err)
+          reject("Got ipfs cat err:" + err)
+        }
+
+        var res = ''
```

Comment:
const?

Reply:
yes again

Reply:
(actually needs to be `let`, as we append to it as chunks come in from ipfs.)